utils_cv/detection/dataset.py

```diff
@@ -223,12 +304,22 @@ def _read_annos(self) -> None:
         self.anno_paths = []
         self.anno_bboxes = []
         self.mask_paths = []
+        self.keypoints = []
         for anno_idx, anno_filename in enumerate(anno_filenames):
             anno_path = self.root / self.anno_dir / str(anno_filename)
 
             # Parse annotation file if present
             if os.path.exists(anno_path):
-                anno_bboxes, im_path = parse_pascal_voc_anno(anno_path)
+                anno_bboxes, im_path, keypoints = parse_pascal_voc_anno(
+                    anno_path, keypoint_meta=self.keypoint_meta
+                )
+                # When meta provided, we assume this is keypoint
+                # detection, and skip the image if no keypoints found.
+                if self.keypoint_meta is not None:
+                    if len(keypoints) != 0:
+                        self.keypoints.append(keypoints)
+                    else:
+                        continue
```

Comment:
So the list self.keypoints could be shorted than self.im_paths, which would be a bug? Should we throw an error here - if the user specifies keypoint_meta, then all images need to have the keypoint annotation for training?

Reply:
Changed.  And the exception will be thrown when parsing the annotation xml file in `utils_cv.detection.dataset.parse_pascal_voc_anno()`